### PR TITLE
sqlbase: plumb DummyPrivilegedAccessor to relevant initializers

### PR DESCRIPTION
**Pull request by otan**
Refs: #45408

No functional change but a help with debugging, initialize
PrivilegedAccessor with a dummy one much like we do with
SessionAccessor.

Release note: None

### CODE REVIEW

File: pkg/sql/planner.go [hunk 1]
```diff
@@ -281,8 +281,8 @@ func newInternalPlanner(
 		ctx, sd, dataMutator, tables, txn, ts, ts, execCfg, &plannerMon,
 	)
 	p.extendedEvalCtx.Planner = p
-	p.extendedEvalCtx.PrivilegedAccessor = p
 	p.extendedEvalCtx.SessionAccessor = p
+	p.extendedEvalCtx.PrivilegedAccessor = p
```
Comment (rohany) [inline pkg/sql/planner.go]: [nit]: unnecessary change

**Comment by rohany** [comment]
lgtm aside from nit

File: pkg/sql/planner.go (hunk 1 above)
Comment (otan) [reply pkg/sql/planner.go]: i'm just tryna keep the order the same hahahah

**Comment by rohany** [comment]
Oh i didn't notice that -- i don't mind either way

**Comment by otan** [comment]
bors r=rohany

**Comment by otan** [comment]
bors r=rohany

**Comment by otan** [comment]
bors r=rohany

Flake with bench again.

**Comment by otan** [comment]
bors r=rohany

Different flake....
